File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            c = (r + l) // 2
            if nums[c] == target:
                return c
            elif nums[c] < target:
                l = c + 1
            elif nums[c] > target:
                r = c - 1
        return -1

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        stable_sort(nums.begin(), nums.end());
        int i = 0;
        int j = nums.size() - 1;
        int sum = 0;
        while (j >= 0 && i < nums.size()) {
            sum = nums[i] + nums[j];
            if (sum == target && i != j) {
                return {i, j};
            } else if (sum > target) {
                j--;
            } else {
                i++;
            }
        }
        return {0,0};
    }
};
